### PR TITLE
fix issue where getMaxAfford didn't check discounted_land

### LINKED ISSUE #442: Discounted acres + discounts should show on construction page
_No content_

---
Beep, boop, I'm [a bot](https://github.com/LeagueSandbox/IssueBot)! This issue was created by `@Sharqy` in `#bug-reports`.

### PR DESCRIPTION
**Pull request by internetfett**
Fixes #442
ConstructionCalculator will now do the discounted land calc as coded in the ConstructActionService.
Also added a note on the construction page template, but not sure about what wording to use:
"Additionally, X acres from invasion can be built at reduced cost."

### CODE REVIEW

File: src/Calculators/Dominion/Actions/ConstructionCalculator.php [hunk 1]
```diff
@@ -130,10 +130,35 @@ public function getLumberCostMultiplier(Dominion $dominion): float
...
+                $maxFromDiscountedPlatinum,
+                $maxFromDiscountedLumber,
+                $dominion->discounted_land,
+                $barrenLand
+            );
+            // Subtract discounted building cost from available resources
+            $platinumToSpend -= (int)ceil(($platinumCost * $discountedBuildings) / 2);
+            $lumberToSpend -= (int)ceil(($lumberCost * $discountedBuildings) / 2);
+            // Avoid division by zero in return statement
+            if($platinumToSpend == 0 || $lumberToSpend == 0) return $discountedBuildings;
```
Comment (stromblom) [inline src/Calculators/Dominion/Actions/ConstructionCalculator.php]: Division by zero works the other way around ;)

**Comment by stromblom** [comment]
Could maybe refactor the discount cost into something reusable and then also refactor it inside ConstructionActionService. As it is now, we have the same calculation done in two different parts of the code.

**Comment by WaveHack** [comment]
Good point.

https://github.com/WaveHack/OpenDominion/blob/develop/src/Services/Dominion/Actions/ConstructActionService.php#L106-L118

This code should probably be refactored in the ConstructorCalculator. Is this something you can pick up for this PR, @internetfett?

**Comment by internetfett** [comment]
> This code should probably be refactored in the ConstructorCalculator. Is this something you can pick up for this PR, @internetfett?

Yeah, I'll work on that ASAP.

**Pull request merged**